blobs/howto/python/python-v12/crud_datalake.py

```diff
@@ -24,287 +24,180 @@
+    # -------------------------------------------------
+    # Create directory method
+    # -------------------------------------------------
 
-# -------------------------------------------------
-# Rename directory method
-# ------------------------------------------------
- 
-# <Snippet_RenameDirectory>
-def rename_directory():
-    try:
-       
-       file_system_client = service_client.get_file_system_client(file_system="my-file-system")
-       directory_client = file_system_client.get_directory_client("my-directory")
-       
-       new_dir_name = "my-directory-renamed"
-       directory_client.rename_directory(new_name=directory_client.file_system_name + '/' + new_dir_name)
-
-    except Exception as e:
-     print(e) 
-# </Snippet_RenameDirectory>
+    # <Snippet_CreateDirectory>
+    def create_directory(self, file_system_client: FileSystemClient, directory_name: str) -> DataLakeDirectoryClient:
+        directory_client = file_system_client.create_directory(directory_name)
 
-# -------------------------------------------------
-# Delete directory method
-# -------------------------------------------------
-
-# <Snippet_DeleteDirectory>
-def delete_directory():
-    try:
-        file_system_client = service_client.get_file_system_client(file_system="my-file-system")
-        directory_client = file_system_client.get_directory_client("my-directory")
+        return directory_client
+    # </Snippet_CreateDirectory>
 
+    # -------------------------------------------------
+    # Rename directory method
+    # ------------------------------------------------
+    
+    # <Snippet_RenameDirectory>
+    def rename_directory(self, directory_client: DataLakeDirectoryClient, new_dir_name: str):
+        directory_client.rename_directory(
+            new_name=f"{directory_client.file_system_name}/{new_dir_name}")
+    # </Snippet_RenameDirectory>
+
+    # -------------------------------------------------
+    # Delete directory method
+    # -------------------------------------------------
+
+    # <Snippet_DeleteDirectory>
+    def delete_directory(self, directory_client: DataLakeDirectoryClient):
         directory_client.delete_directory()
-    except Exception as e:
-     print(e) 
-# </Snippet_DeleteDirectory>
-
-# ------------------------------------------------
-# Upload a file to a directory
-# ------------------------------------------------
-
-# <Snippet_UploadFile>
-def upload_file_to_directory():
-    try:
+    # </Snippet_DeleteDirectory>
 
-        file_system_client = service_client.get_file_system_client(file_system="my-file-system")
+    # ------------------------------------------------
+    # Upload a file to a directory
+    # ------------------------------------------------
 
-        directory_client = file_system_client.get_directory_client("my-directory")
+    # <Snippet_UploadFile>
+    def upload_file_to_directory(self, directory_client: DataLakeDirectoryClient, local_path: str, file_name: str):
```

Comment:
I think we should rename this sample to something like `append_data_to_file` since that's more of the use of the `append_data` API. If you just want to upload a local file to the service, the `upload_data` API would be the recommended way since that is our convenance method that splits the upload into chunks and all. This `append_data` is more commonly used to insert data or append to an existing file.

The below sample can probably then just be called `upload_file_to_directory` since that is the recommended way to do that and I don't think the word "bulk" really means much here. I also think moving the below test above that one is a better logical order since the `upload_data` API is probably our first recommendation and the simpler API.

(This assumes you can do this without breaking articles that link to this, if there are any)

Reply:
Makes sense. I like that idea. I plan to add some additional commentary in the actual article around the differences here, so I'll update the samples as well. Thanks!